addons/edi/models/edi_synchronizer.py

```diff
@@ -370,7 +381,10 @@ def execute(self):
                     vals_list = [rec.target_values(rec._record_values()) for rec in batch]
```

Comment:
My thoughts were about only storing the keys of the fields that change and then update this line to something like:
```
vals_list = [
    {  k: v
        for k,v in rec.target_values(rec._record_values())
        if k in rec.changed_fields
    }
    for rec in batch
    ]
``` 
This way we don't need any of the other changes.